app/CLI/Watch.hs

```diff
@@ -28,19 +34,25 @@ watchAuction auctionName = do
   case mEnhancedTerms of
     Nothing ->
       putStrLn $ "Auction " <> show auctionName <> " does not exist."
-    Just CliEnhancedAuctionTerms {sellerActor} -> do
+    Just eTerms -> do
       currentTime <- getCurrentTime
+      currentTimeMs <- currentTimeMilliseconds
+      currentStage <- currentAuctionStage (terms eTerms)
       let showTime = formatTime defaultTimeLocale "%Y-%m-%d %H:%M:%S"
+          mSecsLeft = secondsLeftInInterval currentTimeMs (stageToInterval (terms eTerms) currentStage)
       putStrLn $ showTime currentTime
       putStrLn $ "Auction: " <> show auctionName
-      putStrLn $ "Seller: " <> show sellerActor
+      putStrLn $ "Seller: " <> show (sellerActor eTerms)
       putStrLn $
         "Current stage: "
-          <> "[TODO: What is the current stage "
-          <> "and how many seconds are left (if applicable)?]"
-      putStrLn $
-        "Standing bid held by: "
-          <> "[TODO: Who submitted the standing bid?]"
+          <> show currentStage
+          <> case mSecsLeft of
+            Nothing -> mempty
+            Just s -> "\n" <> show s <> " seconds left until next stage"
 
   threadDelay 0.2
   watchAuction auctionName
+
+secondsLeftInInterval :: Integer -> Interval POSIXTime -> Maybe Integer
+secondsLeftInInterval now (Interval _ (UpperBound (Finite (POSIXTime t)) inclusive)) = Just $ (t - now - if inclusive then 0 else 1) `div` 1000
```

Comment:
Why you substract 1 but then `div` for 1000? Or div will round that 1 automatically? Probably you right.

Reply:
div already does the rounding. it is what is usually known as integer division